### PR TITLE
add subtitles publishing integration

### LINKED ISSUE #3043: Integration for adding subtitles
Integrate `VideoWorkspace` subtitles support:
- Supply subtitles in the video metadata
- Start subtitles upload
- Show subtitles for existing videos
- Allow downloading subtitles from existing videos
- Allow editing subtitles

### PR DESCRIPTION
**Pull request by kdembler**
**Left to do:** (possibly as a followup)
- Allow downloading already published subtitle (via "Download file" context menu option)
- Improve subtitle validation in VideoForm:
    - Disallow publishing without selecting a subtitle file
    - Disallow selecting subtitle language if subtitles for that language have already been published
- Verify that code I removed from `UploadStatusGroup.tsx` wasn't needed. Confirm that you can still re-upload file from a new browser.

Apart from that everything should be working.

Please reach out to ask any questions or suggest any improvements.

Closes #3043 and #3118

### CODE REVIEW

File: packages/atlas/src/components/_inputs/SubtitlesComboBox/SubtitlesCombobox.tsx
Comment (drillprop) [inline packages/atlas/src/components/_inputs/SubtitlesComboBox/SubtitlesCombobox.tsx]: Hmm, maybe I nitpicking here, but the singular word "subtitle" means: 
1. the title that comes after the main title
2. one line of a movie `subtitles`, I guess?


To avoid some misunderstanding, I think we should always use the plural noun `subtitles`. So in my opinion we should change this to **SubtitlesInput**. The same applies to `SubtitleBox`(I named it wrong when I was working on subtitles UI).
WDYT?

File: packages/atlas/src/providers/videoWorkspace/hooks.ts
Comment (drillprop) [inline packages/atlas/src/providers/videoWorkspace/hooks.ts:43]: hmm, can't we just use `s.type` here?

File: packages/atlas/src/utils/size.ts
Comment (drillprop) [inline packages/atlas/src/utils/size.ts:1]: Is this white space intentional?

File: packages/atlas/src/views/studio/MyUploadsView/UploadStatus/UploadStatus.tsx
Comment (drillprop) [inline packages/atlas/src/views/studio/MyUploadsView/UploadStatus/UploadStatus.tsx]:
```suggestion
      ? 'Subtitles'
```
I would also change `asset.type` as I mentioned in one of my comment above

File: packages/atlas/src/views/studio/MyUploadsView/UploadStatus/UploadStatus.tsx
Comment (drillprop) [inline packages/atlas/src/views/studio/MyUploadsView/UploadStatus/UploadStatus.tsx:287]: image
I'm not sure where we setting these values in the code, but I guess, this should be somewhere in this file. Shouldn't we put subtitles size on the next column(the third one)? Below sizes of the video file and image file?
**Suggestion** - maybe in the second column we could show the language or language iso?

**Review by drillprop — CHANGES_REQUESTED** [review]
Functionality wise works well 👍🏻 , added a few suggestions.

File: packages/atlas/src/providers/videoWorkspace/hooks.ts
Comment (kdembler) [reply packages/atlas/src/providers/videoWorkspace/hooks.ts:43]: Not really, because `s.type` is of type `string`, I need to create additional Typescript guards so we limit the type to `'subtitles' | 'closed-captions'`

File: packages/atlas/src/utils/size.ts
Comment (kdembler) [reply packages/atlas/src/utils/size.ts:1]: Yep, because when we format the size, `10KB` looks good, but `10bytes` is a bit weird

File: packages/atlas/src/providers/videoWorkspace/hooks.ts
Comment (drillprop) [reply packages/atlas/src/providers/videoWorkspace/hooks.ts:43]: What about adding `... as const`? Shouldn't do the job?

Comment (drillprop) [reply packages/atlas/src/providers/videoWorkspace/hooks.ts:43]: nvm, you're right. This will also not work. We're getting this from graphql.

File: packages/atlas/src/components/_inputs/SubtitlesComboBox/SubtitlesCombobox.tsx
Comment (kdembler) [reply packages/atlas/src/components/_inputs/SubtitlesComboBox/SubtitlesCombobox.tsx]: Done, renamed all instances

File: packages/atlas/src/views/studio/MyUploadsView/UploadStatus/UploadStatus.tsx
Comment (kdembler) [reply packages/atlas/src/views/studio/MyUploadsView/UploadStatus/UploadStatus.tsx]: Done

File: packages/atlas/src/views/studio/MyUploadsView/UploadStatus/UploadStatus.tsx
Comment (kdembler) [reply packages/atlas/src/views/studio/MyUploadsView/UploadStatus/UploadStatus.tsx:287]: Good call, added subtitle language in the second column

**Review by drillprop — APPROVED** [review]
Looks good to me 👍🏻 but let's maybe wait for final approval from @rafalpawlow

**Review by rafalpawlow — CHANGES_REQUESTED** [review]
Something is not right, I added 4 different languages but I get doubled `ru` after publish
Zrzut ekranu 2022-08-25 o 20 23 28
Zrzut ekranu 2022-08-25 o 20 25 18

**Comment by kdembler** [comment]
@rafalpawlow This was an initial upload with subtitles or video edit? I can not reproduce this issue, added a bunch of different languages and everything seems to look fine. If you can reproduce, please provide more details.

**Review by rafalpawlow — APPROVED** [review]

**Pull request merged**
